Source/DataSources/KmlDataSource.js

```diff
@@ -1474,27 +1628,370 @@ define([
         console.log('KML - Unsupported feature: ' + node.localName);
     }
 
+    var RefreshMode = {
+        INTERVAL : 0,
+        EXPIRE : 1,
+        STOP : 2
+    };
+
+    var scratchCartesian2_1 = new Cartesian2();
+    var scratchCartesian3_1 = new Cartesian3();
+    var scratchCartesian3_2 = new Cartesian3();
+    var scratchCartesian3_3 = new Cartesian3();
+    var scratchCartesian3_4 = new Cartesian3();
+    var scratchCartographic = new Cartographic();
+    var horizonPoints = [new Cartesian3(), new Cartesian3(), new Cartesian3(), new Cartesian3()];
+
+    function computeHorizonQuad(camera) {
```

Comment:
Both this function and `computeViewRectangle` sound like they should be part of the camera object itself and not specific tot he KML datasource.  Is there a reason why we didn't just add them there?  They also seem like the type of thing you would want to unit test individually.

Is `computeViewRectable` the solution to #871?

CC @bagnell @pjcozzi